canonical-websites/snapcraft.io
# Self-serve categories

Fixes https://github.com/canonical-websites/snapcraft.io/issues/1458

## Done
- Added select boxes and interaction as described in https://docs.google.com/document/d/1Teovi34AVAkHtFdO9agsILqS96QVmaDqZ9DMKyFs48U/edit
- Updated the backend to transform the snap categories to avoid reserved jinja2 key

## QA
- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/snaps
- Select a snap
- You should see a new `Category:` section
- Play around with the categories, adding categories and secondary categories
- You should not be able to select the same category for both the primary and secondary categories
- When you're happy click "Save"
- Visit http://0.0.0.0:8004/search?category=<category_name> for the category(s) you selected
- If you go to the last page you should see your snap (may require patience and refreshes)
- **WHEN DONE, REMOVE THE CATEGORIES FROM TEST SNAPS AS THEY WILL DISPLAY ON PRODUCTION**
